python/deptry/distribution.py

```diff
@@ -0,0 +1,56 @@
+from __future__ import annotations
+
+import re
+from collections import defaultdict
+from functools import lru_cache
+
+from deptry.compat import importlib_metadata
+
+
+@lru_cache(maxsize=None)
+def normalize_distribution_name(name: str) -> str:
+    """
+    Apply name normalization on distribution name, per https://packaging.python.org/en/latest/specifications/name-normalization/#name-normalization.
+    """
+    return re.sub(r"[-_.]+", "-", name).lower()
+
+
+@lru_cache(maxsize=1)
+def get_packages_normalized_distributions() -> dict[str, set[str]]:
```

Comment:
At first glance it is a bit confusing to understand the difference between this function and `get_normalized_distributions_packages`. Maybe we can make their names slightly more verbose? Or add a small example output in their docstring?

```suggestion
def get_packages_to_normalized_distributions_mapping() -> dict[str, set[str]]:
```